src/pint/models/timing_model.py

```diff
@@ -500,20 +499,30 @@ def __getattr__(self, name):
         )
 
     @property_exists
-    def params(self):
-        """List of all parameter names in this model and all its components (order is arbitrary)."""
-        # FIXME: any reason not to just use params_ordered here?
-        p = self.top_level_params
-        for cp in self.components.values():
-            p = p + cp.params
-        return p
+    def params_ordered(self):
+        """List of all parameter names in this model and all its components.
+        This is the same as `params`."""
+
+        # Historically, this was different from `params` because Python
+        # dictionaries were unordered until Python 3.7. Now there is no reason for
+        # them to be different.
+
+        warn(
+            "`TimingModel.params_ordered` is now deprecated and maybe removed in the future. "
```

Comment:
"maybe" -> "may be"

Reply:
also, shouldn't there be some sort of way that the two lines of the warning are connected?  Is this actually tested?

Reply:
Is this best done as a generic `warn` or https://docs.python.org/3/library/exceptions.html#DeprecationWarning?

Reply:
> also, shouldn't there be some sort of way that the two lines of the warning are connected? Is this actually tested?

Python treats such strings as a single string with no newline in between. In a script,
```
"foo"
"bar"
```
is the same as `"foobar"`.

Reply:
> Is this best done as a generic warn or

The `warn` function does invoke a `DeprecationWarning`.

Reply:
ok, I didn't read down far enough.